angr/analyses/reaching_definitions/dep_graph.py

```diff
@@ -16,27 +17,26 @@ class DepGraph:
     Mostly a wrapper around a <networkx.DiGraph>.
     """
 
-    def __init__(self, graph=None):
+    def __init__(self, graph: Optional[networkx.DiGraph]=None):
         """
-        :param networkx.DiGraph graph: A graph where nodes are definitions, and edges represent uses.
+        :param graph: A graph where nodes are definitions, and edges represent uses.
         """
         # Used for memoization of the `transitive_closure` method.
-        self._transitive_closures = {}
+        self._transitive_closures: Dict = {}
 
-        if not isinstance(graph, networkx.DiGraph):
-            self._graph = networkx.DiGraph()
-            return
-
-        if not all(map(_is_definition, graph.nodes)):
+        if graph and not all(map(_is_definition, graph.nodes)):
             raise TypeError("In a DepGraph, nodes need to be <%s>s." % Definition.__name__)
 
-        self._graph = graph
+        if not isinstance(graph, networkx.DiGraph):
```

Comment:
As there is type checking, couldn't this be simplified as:
```
self._graph = graph or networkx.DiGraph()
```
?
(because graph is either a `DiGraph` or `None` at this point)